recipes/log4cxx/all/conanfile.py

```
@@ -12,113 +16,143 @@ class Log4cxxConan(ConanFile):
 
     def requirements(self):
-        self.requires("apr/1.7.0")
         self.requires("apr-util/1.6.1")
-        self.requires("expat/2.4.2")
-        if self.settings.os != "Windows":
-            self.requires("odbc/2.3.9")
+        if self.options.get_safe("with_odbc_appender") and self.settings.os != "Windows":
+            self.requires("odbc/2.3.11")
+        if self.options.get_safe("with_smtp_appender"):
+            self.requires("libesmtp/1.1.0")
+        if self.options.get_safe("with_fmt_layout"):
+            self.requires("fmt/9.1.0")
+        if self.options.get_safe("with_qt"):
+            self.requires("qt/5.15")
 
     @property
     def _compilers_minimum_version(self):
         return {
             "gcc": "7",
             "Visual Studio": "15",
+            "msvc": "191",
             "clang": "5",
             "apple-clang": "10",
         }
 
     def validate(self):
-        # TODO: if compiler doesn't support C++17, boost can be used instead
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "17")
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if not minimum_version:
-            self.output.warn("log4cxx requires C++17. Your compiler is unknown. Assuming it supports C++17.")
-        elif tools.Version(self.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration("log4cxx requires a compiler that supports at least C++17")
-
-    def build_requirements(self):
-        if self.settings.os != "Windows":
-            self.build_requires("pkgconf/1.7.4")
+        if Version(self.version) < "1.0.0" or self.options.get_safe("with_multiprocess_rolling_file_appender"):
+            # TODO: if compiler doesn't support C++17, boost can be used instead
+            if Version(self.version) < "1.0.0":
+                self.output.info(f"Version {self.version} requires C++17. log4cxx version 1.1.0 does not.")
```

is there a reason to only output info? if the user incorrectly configures the graph it should raise an error